framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3240,6 +3240,11 @@ NonlinearSystemBase::computeScalingJacobian(NonlinearImplicitSystem & sys)
     applyScalingFactors(inverse_scaling_factors);
     if (auto displaced_problem = _fe_problem.getDisplacedProblem().get())
       displaced_problem->systemBaseNonlinear().applyScalingFactors(inverse_scaling_factors);
+
+    // Now it's essential that we reset the sparsity pattern of the matrix
+    auto ierr = MatResetPreallocation(petsc_matrix.mat());
```

Comment:
Do you mind to add something like: 

#if LIBMESH_HAVE_PETSC
#endif

Reply:
You need to check PETSC version as well. The API is pretty new.

Reply:
#if !PETSC_VERSION_LESS_THAN(3, 8, 0)
#endif

Reply:
I've already wrapped the entire contents of this function with `LIBMESH_HAVE_PETSC`. With respect to the version, do you think we should just not do scaling if we can't reset the preallocation? Otherwise the first real Jacobian evaluation will be exceedingly slow

Reply:
Right, we should say something like: autoscaling is ignored because you are using too old PETSc. 

I do not like people complain about the performance because of their faults.

Reply:
👍